src/components/PathwayExercise/ContentEdit/index.js

```diff
@@ -0,0 +1,450 @@
+        setCourse(res.data.course.exercises[exerciseId].content);
+        const Date = res.data.course.exercises[exerciseId].updated_at;
+        const date = Date.split("T")[0].replace(
+          /(\d{4})-(\d{1,2})-(\d{1,2})/,
+          function (match, y, m, d) {
+            return d + "/" + m + "/" + y;
+          }
+        );
+        var longDateStr = moment(date, "D/ M/Y").format("dddd D MMMM Y");
+        console.log("longDateStr", longDateStr);
+        setUpdatedOn(longDateStr);
+      })
+      .catch((err) => {
+        console.log("error");
+      });
+  }, [courseId, exerciseId]);
+
+  useEffect(() => {
+    putApiAssessmentCall();
+  }, [save]);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  console.log("course in content edit", course);
+  console.log("id", id);
+
+  return (
+    <>
+      <AppBar fullWidth position="sticky" color="background" elevation={2}>
+        <Box>
+          <Container maxWidth>
+            <Toolbar sx={{ alignItems: "center" }}>
+              <Typography
+                variant="h6"
+                component="div"
+                pt={1}
+                style={{
+                  position: "relative",
+                  left: "-30px",
+                }}
+              >
+                <CloseIcon
+                  onClick={handleClickOpen}
+                  sx={{ color: "#6D6D6D" }}
+                />
+              </Typography>
+              <Box sx={{ flexGrow: 1 }} />
+              <Typography className={classes.edit}>
+                {updatedOn && `Last Edited on ${updatedOn}`}
+              </Typography>
+              <Box sx={{ flexGrow: 1 }} />
+            </Toolbar>
+          </Container>
+        </Box>
+      </AppBar>
+      <Container maxWidth="sm" sx={{ mt: 5, mb: 5 }}>
+        {courseType === "assessment" ? (
+          <>
+            {course &&
+              course.map((e, index) => {
+                if (e.component === "questionExpression") {
+                  console.log(e.component, "e.value", e.value);
+                  return (
+                    <BoxComponent
+                      setIsShown={setIsShown}
+                      isShown={isShown}
+                      iconClick={(e) => handleAdd(index, "assessment")}
+                    >
+                      <Typography>Question</Typography>
+                      <TextareaAutosize
+                        aria-label="empty textarea"
+                        fullWidth
+                        placeholder="Question"
+                        color="primary"
+                        className={classes.textarea}
+                        value={course[index].value}
+                        onChange={(e) => {
+                          var temp = [...course];
+                          temp[index].value = e.target.value;
+                          setCourse(temp);
+                        }}
+                      />
```

Comment:
Duplicated below, factor out into a component? However, for future reference, rather than writing the code first and then trying to factor out common code into components, it'll be good to start with a mock and then break it into a hierarchy as outlined here: https://reactjs.org/docs/thinking-in-react.html#start-with-a-mock. The point is that components should be small and have only one reason to change to maximize reusability and make them easier to test and debug.